packages/react-ui/src/custom-components/InputFile.js

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Box, Button, makeStyles, Typography } from '@material-ui/core';
+
+const useStyles = makeStyles((theme) => ({
+  inputFile: {
+    flex: 1,
+    width: '100%'
+  },
+  inputWrapper: {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    flex: 1,
+    height: theme.spacing(4.5),
+    marginRight: theme.spacing(1.5),
+    padding: theme.spacing(0, 1),
+    border: `2px solid ${theme.palette.action.disabled}`,
+    borderRadius: theme.spacing(0.5),
+    transition: theme.transitions.create(['border'], {
+      duration: theme.transitions.duration.short,
+      easing: theme.transitions.easing.easeInOut
+    })
+  },
+  inputWrapperOver: {
+    border: `2px solid ${theme.palette.primary.light}`
+  },
+  placeholder: {
+    opacity: 0.42,
+    maxWidth: 'calc(100% - 110px)'
+  },
+  placeholderFiles: {
+    opacity: 1
+  }
+}));
+
+function InputFile(props) {
+  const classes = useStyles();
+  const uploadInputRef = useRef(null);
+  const [filesText, setFilesText] = useState('');
+  const [over, setOver] = useState(false);
+
+  useEffect(() => {
+    if (props.files.length === 0) {
+      setFilesText('');
+    } else if (props.files.length === 1) {
+      setFilesText(props.files[0].name);
+    } else {
+      setFilesText(`${props.files.length} files selected`);
+    }
+  }, [props.files]);
+
+  const handleBrowse = () => {
+    uploadInputRef.current?.click();
+  };
+
+  const handleDragOver = (event) => {
+    event.preventDefault();
+    setOver(true);
+  };
+
+  const handleDragLeave = (event) => {
+    event.preventDefault();
+    setOver(false);
+  };
+
+  const handleDrop = (event) => {
+    event.preventDefault();
+
+    const newFiles = [];
+    const items = event.dataTransfer.items;
+
+    for (let i = 0; i < items.length; i++) {
```

Comment:
Why are you not using a forEach instead?

I would change it for:

```ts
items.forEach(item => { .... })
```

Also I think that would be better to move this block to a new function, something like that:
```ts
// handleDrop function
const newFiles = getAllFiles(items)
props.onChange(newFiles)
```

Reply:
`items` is not an array, so instead using `forEach` I'm going to use a `for(item of items)`